docs/source/conf.py

```diff
@@ -54,22 +50,38 @@
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ["_templates"]
 
+# Settings for autodoc
+autodoc_typehints = "description"
+autodoc_typehints_description_target = "all"
+autodoc_typehints_format = "short"
+autodoc_type_aliases = {
+    "ShapeType": "typing.ShapeType",
+    "ScalarType": "typing.ScalarType",
+    "MatrixType": "typing.MatrixType",
+    "IntLike": "typing.IntLike",
+    "FloatLike": "typing.FloatLike",
+    "ShapeLike": "typing.ShapeLike",
+    "DTypeLike": "typing.DTypeLike",
+    "ArrayIndicesLike": "typing.ArrayIndicesLike",
+    "ScalarLike": "typing.ScalarLike",
+    "ArrayLike": "typing.ArrayLike",
+    "LinearOperatorLike": "typing.LinearOperatorLike",
+    "NotImplementedType": "typing.NotImplementedType",
+}
```

Comment:
```suggestion
autodoc_type_aliases = {type_alias: f"typing.{type_alias}" for type_alias in pn.typing.__all__}
```

and then adding the type aliases to `pn.typing.__all__` might scale a little bit better

Reply:
oh yeah this is much better. :+1: